Add InheritedWidget for send info to children classes.

MaksTsymbal opened the pull request:
Add basic UserInfo page.
Add some updating state widgets.
Change README.

MaksTsymbal committed:
Add InheritedWidget for send info to children classes.
Add basic UserInfo page.
Add some updating state widgets.
Change README.

VadymPinchuk reviewed (changes requested):
1. Деякий код уже був. Це надлишково і дуже заважає
2. по неймінгу не бачу змін. ТАкі речі як createDocumentPage  - мають називатись create_document_page - за такі речі буду знімати бали
3. не бачу нічого про InheritedWidget  - як було у ДЗ (Урок 12)
4. Є службові файли які заважають. за таке також буду знімати бали


Здається що з попередній код рев.ю не було зроблено висновків, тому мушу повернути на допрацювання. Наступний раз буде фінальним

MaksTsymbal committed:
Rename all dir
Add more implementation for InheritedWidget
Now when you want to create doc, you will see main section with some widgets, when you pick info in this widgets ang go to table section InheritedWidget helps to view all your changesfrom parrents pages

VadymPinchuk commented on lib/src/create_document_page/ui/create_document_page.dart:
```diff
@@ -0,0 +1,115 @@
+
+  @override
+  Widget build(BuildContext context) {
+    final info = MyInheritedWidget.of(context);
```
він сам себе ініціює тепер? 
Тут додаток одразу має звалитись, оскільки ми не можемо отримати доступ до віджета. який ми вставляємо в дерево віджетів одною строкою нижче

VadymPinchuk commented on lib/src/main_page/data/inherited_widget.dart:
```diff
@@ -0,0 +1,73 @@
+  }) {
+    final widget = MyInheritedWidget.of(context);
+
+    Navigator.of(context).pushReplacement(
```
не можна так ) 

Я давав приклад. Віджет не має сам себе перевставляти в дерево.
Віджет має бути в дереві. у нього має бути стан. цей стан буде джерелом правди. Стан може бути змінним але віджет не може змінюватись. Він має залишатись єдиним

MaksTsymbal committed:
Fix code after review
All work, i add gif

MaksTsymbal committed:
Push new code
With Bad InheritedWidget
